src/app/components/views/login-view/login-view.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { CookieService } from 'ngx-cookie-service';
+
+@Component({
+  selector: 'app-login-view',
+  templateUrl: './login-view.component.html',
+  styleUrls: ['./login-view.component.scss']
+})
+export class LoginViewComponent implements OnInit {
+
+  public uidForm!: FormGroup; // The form for logging in to ABC
+
+  constructor(private cookieService: CookieService, private router: Router) { }
+
+  ngOnInit(): void {
+
+    this.uidForm = new FormGroup({
+      'uid': new FormControl(null, [
+        Validators.required,
+        Validators.minLength(6),
+        Validators.maxLength(6),
+        Validators.pattern(/[a-z]+[a-z]+[0-9]+[0-9]+[0-9]+[a-z0-9]/) // standard UID formatting
+      ]),
+    });
+
+    this.checkLogin()
+  };
+
+  /**
+   * @description Saving login information when information is entered then kicks you home
+   * @returns {void}
+   */
+  public onLogin(): void {
+    const userLogin = this.uidForm?.get('uid')?.value;
```

Comment:
Set data type for user Login

Also.  Theres random use of optional chainong here.

Reply:
for atlas standards, we dont use the optional chaining inside the ts file....i dont have a good reason why, but we dont...just so youre aware